Global/Metals.gitignore

```diff
@@ -0,0 +1,4 @@
+# Metals specific
```

Comment:
Is it worth writing a bit in the header here about what Metals is, why it's relevant, and linking to the project page? These templates are consumed in many unexpected places, and this context often helps users understand the template when they stumble upon it and they're far removed from the original PR.

The example that I keep pointing people too is probably `JetBrains.gitignore`:

https://github.com/github/gitignore/blob/7b5783ace752279104f7bf7a812c6f6225f30cfd/Global/JetBrains.gitignore#L1-L4

Reply:
Sure, I've added in a short description and a link to the Metals website.